src/main/java/FileUtils.java

```diff
@@ -0,0 +1,54 @@
+import java.io.*;
+import java.util.Scanner;
+
+public class FileUtils {
+    static MathematicalOperation[] read(String fileName) throws FileNotFoundException {
+        Scanner scanner = new Scanner(new File(fileName));
+        int lines = countLines(fileName);
+        MathematicalOperation[] result = new MathematicalOperation[lines];
+        for (int i = 0; i < lines; i++) {
+            String line = scanner.nextLine();
+            String[] split = line.split(" ");
+            double firstNumber = Double.parseDouble(split[0]);
+            double secondNumber = Double.parseDouble(split[2]);
+            Operators operator = setOperator(split[1]);
+            double price = Double.parseDouble(split[2]);
+            result[i] = new MathematicalOperation(firstNumber, secondNumber, operator);
+        }
+        return result;
+    }
+
+    static void save(MathematicalOperation[] mathematicalOperations, String fileName) throws IOException {
+        BufferedWriter writer = new BufferedWriter(new FileWriter(fileName));
+        for (MathematicalOperation mathematicalOperation : mathematicalOperations) {
+            writer.write(mathematicalOperation.toString());
+            writer.newLine();
+        }
+        writer.close();
```

Comment:
Co prawda zamykasz ten plik później, ale w przypadku błędu może pozostać niezamknięty. Najlepiej wprowadź try-with-resources https://youtu.be/LRwSHA9YDgY